WindowServer: Fix various annoying things

BenWiederhake opened the pull request:
In particular:
- Callers of `WindowManager::desktop_rect()` might have misbehaved during fullscreen (undocumented issue)
- Fix required visible region at the bottom (undocumented issue)
- Allow superdrags to hide the titlebar. #5105
- Fixes WindowServer crashing when a malicious program creates a window of size (0, 0) (undocumented issue)
- Constrain geometry label on move/resize to desktop #5063

I didn't know that other people were working on this, too D:

- Might conflict with PR #5130, if it fixes the issues around setting/reading `m_unmaximized_rect`
- Does *not* conflict with PR #5138
- Does *not* conflict with PR #5129

nico commented on Userland/Services/WindowServer/WindowManager.cpp:
```diff
@@ -1052,6 +1052,25 @@ Gfx::IntRect WindowManager::desktop_rect() const
     };
 }
 
+Gfx::IntRect WindowManager::arena_rect_for_type(WindowType type) const
```
s/arena/area/ (?)

nico replied:
But even then I don't get what the name is supposed to mean. "screen_rect_for_window_type" (?)

BenWiederhake replied:
That was not a typo: "Arena" as in "the space where you can move around". I'm not sure whether `screen_rect` is more intuitive than `arena`, but at least it looks less like a typo.

emanuele6 commented on Userland/Services/WindowServer/WindowManager.cpp:
```diff
@@ -554,8 +554,10 @@ bool WindowManager::process_ongoing_window_move(MouseEvent& event, Window*& hove
                 m_move_window->set_position_without_repaint(pos);
-                // "Bounce back" the window it it would end up too far outside the screen:
-                m_move_window->normalize_rect();
+                // "Bounce back" the window it it would end up too far outside the screen.
```
> it it

BenWiederhake replied:
Thanks!

BenWiederhake commented:
Changes since last push:
- Fixed a typo I did in FIXME, thanks to @emanuele6
- Fixed a typo in one of the commit messages themselves
- Rebased for convenience

I decided against @nico's suggestion to rename from "arena" to "screen_rect", as it does not refer to the entire screen, and the name is also not a typo.
